app/src/state/server.ts

```diff
@@ -1,169 +1,53 @@
-  if (proc == undefined) {
-    return;
-  }
-  dispatch(setLocalProc(proc));
-  dispatch(setLocalState(LocalServerStatus.Starting));
-  proc.stdout.on('data', (data: Buffer) => {
-    console.log(`Server stdout: ${data}`);
-    try {
-      const parsed_data: ServerStartingMessage | ServerStartedMessage = JSON.parse(data.toString());
-      if (parsed_data.msg == 'server_starting') {
-        dispatch(setLocalPort(parsed_data['port']));
-      } else if (parsed_data.msg == 'server_started') {
-        dispatch(setLocalToken(parsed_data['token']));
-        dispatch(setLocalState(LocalServerStatus.Running));
-      }
-    } catch (e) {
-      console.log('error decoding stdout json', e);
-    }
-  });
-
-  proc.stderr.on('data', (data: Buffer) => {
-    console.error(`Server stderr: ${data}`);
-  });
 
-  proc.on('close', (code: number | null) => {
-    if (code == 0) {
-      dispatch(setLocalState(LocalServerStatus.Stopped));
-    } else {
-      dispatch(setLocalState(LocalServerStatus.Error));
-    }
-    dispatch(setLocalProc(null));
-    console.log(`child process exited with code ${code}`);
-  });
-});
-
-export const stopServer = createAsyncThunk<void, void, { state: RootState }>(
-  'server/stopServer',
-  async (_, { dispatch, getState }) => {
-    const proc = getState().server.local_proc;
-    if (proc) {
-      console.log('killing proc', proc);
-      proc.kill();
-      dispatch(setLocalState(LocalServerStatus.Stopping));
-    }
+export const changeServer = createAsyncThunk<void, number, { state: RootState }>(
+  'models/changeServer',
+  async (server, { dispatch }) => {
+    await dispatch(selectServer(server));
+    dispatch(fetchModelState());
   }
 );
 
 export const serverSlice = createSlice({
   name: 'server',
   initialState: {
-    local_state: LocalServerStatus.NotStarted,
-    local_proc: null,
-    local_config: {
-      hostname: 'http://localhost',
-      port: 0,
-      name: 'Local Server',
-      token: null,
-    },
     servers: [],
-    selected_server: null,
+    selectedServer: 0,
   } as ServerState,
   reducers: {
-    setLocalState: (state, args: PayloadAction<LocalServerStatus>) => {
-      state.local_state = args.payload;
-    },
-    setLocalProc: (state, args: PayloadAction<ChildProcess | null>) => {
-      state.local_proc = args.payload;
-    },
-    setLocalPort: (state, args: PayloadAction<number>) => {
-      state.local_config.port = args.payload;
-    },
-    setLocalToken: (state, args: PayloadAction<string>) => {
-      state.local_config.token = args.payload;
-    },
     addServer: (state, args: PayloadAction<ServerConfig>) => {
       state.servers.push(args.payload);
     },
+    setLocalServer: (state, args: PayloadAction<Omit<ServerConfig, 'name'>>) => {
+      const localServerName = 'local server';
+      const otherServers = state.servers.filter((server) => server.name != localServerName);
+      state.servers = [{ name: localServerName, ...args.payload }, ...otherServers];
```

Comment:
This could be dangerous in the future when we support more than the local server, as it could shift which server is selected (e.g. we don't have a local server, but two remote (A and B). you select B [state.selectedServer = 1]. then we add the local server and now A in selected)